src/org/zaproxy/zap/extension/websocket/ExtensionWebSocket.java

```diff
@@ -290,6 +293,10 @@ public void databaseOpen(Database db) throws DatabaseException, DatabaseUnsuppor
 	public void hook(ExtensionHook extensionHook) {
 		super.hook(extensionHook);
 		
+		API.getInstance().registerApiImplementor(api);
+
+		extensionHook.addApiImplementor(api);
```

Comment:
Shouldn’t there be something in the unload method to remove this?

Reply:
When added with `extensionHook` the components are automatically removed/unloaded. The `API.getInstance().registerApiImplementor(api);` should be removed though (redundant with this statement).